### PR TITLE
Feature/agent manager chat mode

### PR DESCRIPTION
**Pull request by almog100**
Chat with text only  mode: chatWithoutStream

### CODE REVIEW

File: src/types/entities/agents/manager.ts [hunk 1]
```diff
@@ -102,12 +102,16 @@ export interface AgentManager {
      * Method to close all connections with agent, stream and web socket
      */
     disconnect: () => Promise<void>;
+    /**
```
Comment (benyael15) [inline src/types/entities/agents/manager.ts:105]: wouldnt it be easier and cleaner to do some like

chat: (userMessage: string, chatMode: chat-mpode,append_chat?: boolean)

within the same methos?

File: src/createAgentManager.ts [hunk 2]
```diff
@@ -252,7 +255,70 @@ export async function createAgentManager(agent: string, options: AgentManagerOpt
 
             analytics.track('agent-chat', { event: 'disconnect', chatId: items.chat?.id, agentId: agentInstance.id });
         },
-        async chat(userMessage: string, append_chat: boolean = false, enforceTextOnly: boolean = false) {
+        async chatWithoutStream(userMessage: string) {
+            try {
```
Comment (benyael15) [inline src/createAgentManager.ts:259]: add chat_mode to all MP analytics

File: src/types/entities/agents/manager.ts (hunk 1 above)
Comment (almog100) [reply src/types/entities/agents/manager.ts:105]: we wanted to handle chatMode only in agents-ui, so we separate everything in agent-sdk so things won't get mixed up and only in agents-ui I have this differentiation by chatMode.

Comment (benyael15) [reply src/types/entities/agents/manager.ts:105]: gotchta - this is sdk :)

File: infra/build.js [hunk 3]
```diff
@@ -36,6 +36,8 @@ try {
     console.log('start copy');
     const root = path.resolve(import.meta.url, '../../').split(':')[1];
     const dist = path.resolve(root, './dist');
+    const packageJson = path.resolve(root, './package.json');
+    console.log(packageJson);
```
Comment (benyael15) [inline infra/build.js:40]: remove log

File: infra/build.js [hunk 4]
```diff
@@ -52,6 +54,7 @@ try {
 
     console.log('Copying new package');
     fs.cpSync(dist, embeddedSdk, { recursive: true });
+    fs.copyFileSync(packageJson, path.resolve(embeddedSdk, '../package.json'));
 } catch (e) {
-    console.log('Copy failed');
+    console.log('Copy failed', e);
```
Comment (benyael15) [inline infra/build.js:59]: remove

**Review by benyael15 — APPROVED** [review]

File: infra/build.js (hunk 4 above)
Comment (HaMMeRSI) [reply infra/build.js]: this is the build script

**Pull request merged**